### PR TITLE
Document Aggregate::toIntermediate() API

### PR DESCRIPTION
**Pull request by mbasmanova**
Extend How to add an aggregate function? documentation article to describe the 
Aggregate::toIntermediate() API.

### CODE REVIEW

**Review by amitkdutta — APPROVED** [review]
Thanks @mbasmanova

**Review by kagamiori — APPROVED** [review]
LGTM. Thanks!

File: velox/docs/develop/aggregate-functions.rst [hunk 1]
```diff
@@ -332,6 +353,43 @@ Finally, we implement the addSingleGroupIntermediateResults() method that is use
           const std::vector<VectorPtr>& args,
           bool mayPushdown) = 0;
 
+Finally, we can implement optional methods for converting raw inputs into intermediate
```
Comment (pedroerp) [inline velox/docs/develop/aggregate-functions.rst:356]: there's another "finally" in line 336. Maybe remove the previous one to avoid repetition?

**Review by pedroerp — APPROVED** [review]
Thanks for improving our docs!

**Pull request closed without merging**